Fix: Clarify line breaks in object-curly-newline (fixes #14024) (#14063)
eslint/eslint: docs/rules/object-curly-newline.md

> # enforce consistent line breaks after opening and before closing braces (object-curly-newline)

Author: Right now this is similar to https://eslint.org/docs/rules/array-bracket-newline.
I think both of them should be written the same way.
I can change this one and make another PR for array-bracket-newline. WDYT?

> # enforce consistent line breaks after opening and before closing braces (object-curly-newline)

Reviewer: Sorry for the delay. Yes, sounds like a good plan!

A number of style guides require or disallow line breaks inside of object braces and other tokens.

## Rule Details

> This rule enforces consistent line breaks after opening and before closing braces of object literals or destructuring assignments.

Author: What about **replacing** 

> This rule enforces consistent line breaks after opening and before closing braces of object literals or destructuring assignments

with:
> More precisely, the rule requires or disallows a line break between { and its following token, and between } and its preceding token

?

> This rule enforces consistent line breaks after opening and before closing braces of object literals or destructuring assignments.

Reviewer: Makes sense to merge these two sentences into one if it would be more understandable.

# enforce consistent line breaks after opening and before closing braces (object-curly-newline)

A number of style guides require or disallow line breaks inside of object braces and other tokens.

## Rule Details

> More precisely, the rule requires or disallows a line break between { and its following token, and between } and its preceding token of object literals or destructuring assignments.

Reviewer: ```suggestion
This rule requires or disallows a line break between `{` and its following token, and between `}` and its preceding token of object literals or destructuring assignments.
```

Since this is now the first sentence in this section, it wouldn't be clear what "More precisely" refers to.

Also added backticks around `{` and `}`
